// @awesome-cordova-plugins/printer/index.js
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
import { AwesomeCordovaNativePlugin, cordova, checkAvailability, getPromise } from '@awesome-cordova-plugins/core';
var PrinterOriginal = /** @class */ (function (_super) {
    __extends(PrinterOriginal, _super);
    function PrinterOriginal() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    /**
     * Checks whether the device is capable of printing (uses `check()` internally)
     *
     * @returns {Promise<boolean>}
     */
    PrinterOriginal.prototype.isAvailable = function () {
        return this.check().then(function (res) { return Promise.resolve(res.avail); });
    };
    PrinterOriginal.prototype.check = function () {
        var _this = this;
        return (function () {
            if (checkAvailability(_this) === true) {
                return getPromise(function (resolve) {
                    PrinterOriginal.getPlugin().canPrintItem(function (avail, count) {
                        resolve({ avail: avail, count: count });
                    });
                });
            }
        })();
    };
    PrinterOriginal.prototype.pick = function () { return cordova(this, "pick", {}, arguments); };
    PrinterOriginal.prototype.print = function (content, options) { return cordova(this, "print", { "successIndex": 2, "errorIndex": 4 }, arguments); };
    PrinterOriginal.pluginName = "Printer";
    PrinterOriginal.plugin = "cordova-plugin-printer";
    PrinterOriginal.pluginRef = "cordova.plugins.printer";
    PrinterOriginal.repo = "https://github.com/katzer/cordova-plugin-printer";
    PrinterOriginal.platforms = ["Android", "iOS", "Windows"];
    return PrinterOriginal;
}(AwesomeCordovaNativePlugin));
var Printer = new PrinterOriginal();
export { Printer };
