// @ionic-native/app-version/__ivy_ngcc__/ngx/index.js
import { __extends } from "tslib";
import { Injectable } from '@angular/core';
import { IonicNativePlugin, cordova } from '@ionic-native/core';
import * as ɵngcc0 from '@angular/core';
var AppVersion = /** @class */ (function (_super) {
    __extends(AppVersion, _super);
    function AppVersion() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    AppVersion.prototype.getAppName = function () { return cordova(this, "getAppName", {}, arguments); };
    AppVersion.prototype.getPackageName = function () { return cordova(this, "getPackageName", {}, arguments); };
    AppVersion.prototype.getVersionCode = function () { return cordova(this, "getVersionCode", {}, arguments); };
    AppVersion.prototype.getVersionNumber = function () { return cordova(this, "getVersionNumber", {}, arguments); };
    AppVersion.pluginName = "AppVersion";
    AppVersion.plugin = "cordova-plugin-app-version";
    AppVersion.pluginRef = "cordova.getAppVersion";
    AppVersion.repo = "https://github.com/whiteoctober/cordova-plugin-app-version";
    AppVersion.platforms = ["Android", "iOS", "Windows"];
AppVersion.ɵfac = /*@__PURE__*/ function () { var ɵAppVersion_BaseFactory; return function AppVersion_Factory(t) { return (ɵAppVersion_BaseFactory || (ɵAppVersion_BaseFactory = ɵngcc0.ɵɵgetInheritedFactory(AppVersion)))(t || AppVersion); }; }();
AppVersion.ɵprov = /*@__PURE__*/ ɵngcc0.ɵɵdefineInjectable({ token: AppVersion, factory: function (t) { return AppVersion.ɵfac(t); } });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(AppVersion, [{
        type: Injectable
    }], null, null); })();
    return AppVersion;
}(IonicNativePlugin));
export { AppVersion };
